components/RequestLogsDashboard.js

```diff
@@ -1,113 +1,68 @@
-import moment from 'moment'
-import { Component } from 'react'
-import { Button } from 'react-bootstrap'
-import { withAuth } from 'use-auth0-hooks'
+import useSWR from 'swr'
+import { useAuth } from 'use-auth0-hooks'
 
 import ApiKeyUsage from './ApiKeyUsage'
-import { secureFetch } from '../util/middleware'
+import FetchMessage from './FetchMessage'
+import { AUTH0_SCOPE } from '../util/constants'
 
-class RequestLogsDashboard extends Component {
-  constructor (props) {
-    super(props)
-    this.state = {
-      logs: null,
-      logsError: null
-    }
-  }
-
-  handleFetchLogs = async (force = false) => {
-    const { logs, logsError } = this.state
-    if (!force && (logs || logsError)) {
-      return
-    }
-
-    const { accessToken } = this.props.auth
-    if (!accessToken) {
-      return
-    }
-    const fetchedLogs = await secureFetch(`${process.env.API_BASE_URL}/api/secure/logs`, accessToken)
-
-    if (fetchedLogs.status === 'success') {
-      this.setState({
-        logs: fetchedLogs.data,
-        fetchMessage: `Updated at ${moment().format('h:mm:ss a')}`
-      })
-    } else {
-      this.setState({
-        fetchMessage: `Failed to update!`
-      })
-    }
-  }
-
-  async componentDidMount () {
-    await this.handleFetchLogs()
-  }
-
-  render () {
-    const { auth, isAdmin } = this.props
-    const { logs, logsError, fetchMessage } = this.state
-    if (!auth.isAuthenticated) return null
-    return (
-      <div>
-        <h2>Request Log Summary</h2>
-        <div className='controls'>
-          <Button onClick={this.handleFetchLogs}>
-            Fetch logs
-          </Button>
```

Comment:
This button to refetch is gone. Granted, with SWR, it gets automatically refetched every 30 seconds, but would it be useful to have a way to refetch immediately or at least show that refetching will occur in x# of seconds?